packages/api/amplify_api/lib/src/graphql/web_socket/types/connectivity_status.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Possible network connectivity statuses.
+enum ConnectivityStatus {
+  /// Network connection detected.
+  connected,
+
+  /// Loss of network connection detected.
+  disconnected,
+}
+
+/// {@template amplify_api_dart.connectivity_interface}
+/// Used to create a stream representing network connectivity at the hardware level.
+/// {@endtemplate}
+abstract class ConnectivityInterface {
+  /// {@macro amplify_api_dart.connectivity_interface}
+  const ConnectivityInterface();
+
+  /// Generates a new stream of [ConnectivityStatus].
+  Stream<ConnectivityStatus> get onConnectivityChanged;
```

Comment:
Since it's not really an interface if you do that - maybe `ConnectivityPlatform` is more appropriate? I'm trying to avoid a `null` value for this anywhere.

Reply:
+1

Reply:
`ConnectivityPlatform` seems like a fine name. I just pushed a commit with this change and other comments addressed.